src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,460 @@
+    function maxWithdraw() external view returns (uint256);
+    function active() external view returns (bool);
+}
+
+interface D3MPlanLike {
+    function getTargetAssets(uint256) external view returns (uint256);
+    function active() external view returns (bool);
+}
+
+interface VatLike {
+    function debt() external view returns (uint256);
+    function hope(address) external;
+    function ilks(bytes32) external view returns (uint256, uint256, uint256, uint256, uint256);
+    function Line() external view returns (uint256);
+    function urns(bytes32, address) external view returns (uint256, uint256);
+    function gem(bytes32, address) external view returns (uint256);
+    function live() external view returns (uint256);
+    function slip(bytes32, address, int256) external;
+    function move(address, address, uint256) external;
+    function frob(bytes32, address, address, address, int256, int256) external;
+    function grab(bytes32, address, address, address, int256, int256) external;
+    function fork(bytes32, address, address, int256, int256) external;
+    function suck(address, address, uint256) external;
+}
+
+interface EndLike {
+    function debt() external view returns (uint256);
+    function skim(bytes32, address) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (address);
+    function join(address, uint256) external;
+    function exit(address, uint256) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external returns (bool);
+}
+
+contract DssDirectDepositHub {
+
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssDirectDepositHub/not-authorized");
+        _;
+    }
+
+    enum Mode{ NORMAL, MODULE_CULLED, MCD_CAGED }
+    uint256             constant  RAY  = 10 ** 27;
+
+    VatLike      public immutable vat;
+    DaiJoinLike  public immutable daiJoin;
+    address      public           vow;
+    EndLike      public           end;
+
+    struct Ilk {
+        D3MPoolLike pool;   // Access external pool and holds balances
+        D3MPlanLike plan;   // How we calculate target debt
+        uint256     tau;    // Time until you can write off the debt [sec]
+        uint256     culled; // Debt write off triggered
+        uint256     tic;    // Timestamp when the pool is caged
+    }
+    mapping (bytes32 => Ilk) public ilks;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed ilk, bytes32 indexed what, address data);
+    event File(bytes32 indexed ilk, bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
```

Comment:
Move this up a couple of lines to match the ordering of these file functions below.

Reply:
done: 22c50f5

Reply:
done 22c50f5